spring-integration-core/src/main/java/org/springframework/integration/dsl/context/IntegrationFlowBeanPostProcessor.java

```diff
@@ -408,8 +408,7 @@ private boolean noBeanPresentForComponent(Object instance, String parentBeanName
 						&& !instance.equals(this.beanFactory.getBean(beanName))) {
 
 					AbstractBeanDefinition beanDefinition =
-							BeanDefinitionBuilder.genericBeanDefinition((Class<Object>) instance.getClass(),
-											() -> instance)
+							BeanDefinitionBuilder.genericBeanDefinition(instance.getClass())
```

Comment:
Revert this one. You essentially making a bug here: a bean definition is going to be based on a new instance of the class, but not provided one as intention.
More over this class does its stuff at runtime, so the change is redundant.